httpx/_client.py

```diff
@@ -793,6 +807,22 @@ def _send_handling_auth(
                 request = next_request
                 history.append(response)
 
+    def _send_handling_retries(self, request: Request, timeout: Timeout) -> Response:
+        retries_left = self.retries
+        delays = exponential_backoff(factor=RETRIES_BACKOFF_FACTOR)
+
+        while True:
+            try:
+                return self._send_single_request(request, timeout)
+            except (ConnectError, ConnectTimeout):
+                if request.method not in IDEMPOTENT_METHODS:
```

Comment:
I don't think we need to care about idempotency for connect retries.
If we've not established a connection to the server, then we're okay to re-issue the request *whatever* the method.

Eg. see connect retries vs. read retries behaviour in urllib3 https://github.com/urllib3/urllib3/blob/f0c43419e45db17a3c5f287be80aa0d14e2f58c6/src/urllib3/util/retry.py#L415-L427